src/components/categoryFilter/CategoryFilter.tsx

```diff
@@ -49,13 +49,21 @@ export const CategoryFilter = <T,>({
     setSelected(calculateInitialSelected(variant, categories))
   }, [variant, categories])
 
+  useEffect(() => {
+    // When all checkbox options are unselected, revert to selecting all
+    if (variant === 'checkbox' && !selected.length) {
+      setSelectAll(true)
+    }
+  }, [variant, selected])
+
```

Comment:
This will need to be covered with a unit test.

Reply:
I think I've added this in CategoryFilter.test I also had to change the helpers.test because when I ran all tests they failed because the dates were being transformed into locale strings.